src/main/java/org/mockito/internal/creation/instance/ConstructorInstantiator.java

```diff
@@ -47,18 +51,11 @@ public ConstructorInstantiator(Object outerClassInstance) {
         return (T) constructor.newInstance(params);
     }
 
-    private static <T> InstantiationException paramsException(Class<T> cls, Exception cause) {
-        return new InstantiationException(
-                join("Unable to create instance of '" + cls.getSimpleName() + "'.",
-                     "Please ensure that the outer instance has correct type and that the target class has 0-arg constructor."),
-                cause);
-    }
-
-    private static <T> InstantiationException noMatchingConstructor(Class<T> cls) {
-        return new InstantiationException(
-                join("Unable to create instance of '" + cls.getSimpleName() + "'.",
-                     "Unable to find a matching 1-arg constructor for the outer instance.")
-                , null);
+    private <T> InstantiationException paramsException(Class<T> cls, Exception e) {
+        return new InstantiationException(join(
+                "Unable to create instance of '" + cls.getSimpleName() + "'.",
+                "Please ensure that it has a constructor that matches these arguments: " + Arrays.toString(constructorArgs))
+                , e);
     }
 
     private static boolean paramsMatch(Class<?>[] types, Object[] params) {
```

Comment:
I am not sure if this alogrithm here works as expected. The JVM would choose the most specific constructor, but we choose the first matching. This can cause some trouble that is hard to detect. 

Assume we have two `String` arguments and get constructor's in this order:
1. MyClass(Object a, Object b)
2. MyClass(String a, String b)

Since `String` is an instance of `Object `we would choose the first constructor, whereas the JVM would choose the second. This should be fixed!